CMakeLists.txt

```diff
@@ -373,10 +373,13 @@ if(VLLM_GPU_LANG STREQUAL "CUDA")
     set_gencode_flags_for_srcs(
       SRCS "${MARLIN_TEMPLATE_KERNEL_SRC}"
       CUDA_ARCHS "${MARLIN_ARCHS}")
-    if(${CMAKE_CUDA_COMPILER_VERSION} VERSION_GREATER_EQUAL 12.8)
-      set_source_files_properties(${MARLIN_TEMPLATE_KERNEL_SRC}
-        PROPERTIES COMPILE_FLAGS "-static-global-template-stub=false")
-    endif()
+    # We must set -static-global-template-stub=false due to referencing
+    # kernel templates instantiated in other compilation units.
+    set_property(
+      SOURCE ${MARLIN_TEMPLATE_KERNEL_SRC}
+      APPEND PROPERTY
+      COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:CUDA>:-static-global-template-stub=false>"
+    )
```

Comment:
Please explain this change?  (and the similar-looking one below around like 962)

Reply:
Actually not needed as it was already resolved in https://github.com/vllm-project/vllm/commit/3e6dd400160525b71b19e3f94b0066b1bea7bb97